bot/src/main/java/com/dongtronic/diabot/platforms/discord/commands/quote/QuoteSearchCommand.kt

```diff
@@ -0,0 +1,119 @@
+package com.dongtronic.diabot.platforms.discord.commands.quote
+
+import com.dongtronic.diabot.util.Patterns
+import com.dongtronic.diabot.data.mongodb.QuoteDAO
+import com.dongtronic.diabot.data.mongodb.QuoteDTO
+import com.dongtronic.diabot.platforms.discord.commands.DiscordCommand
+import com.jagrosh.jdautilities.command.CommandEvent
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.launch
+import org.litote.kmongo.regex
+import org.litote.kmongo.and
+import org.bson.conversions.Bson
+import net.dv8tion.jda.api.EmbedBuilder
+
+class QuoteSearchCommand(category: Category, parent: QuoteCommand) : DiscordCommand(category, parent) {
+    private val maxNumberOfQuotes: Int
+
+    init {
+        this.name = "search"
+        this.help = "Searches through quotes for keywords (quotes contain *all* key words)"
+        this.guildOnly = true
+        this.aliases = arrayOf("s")
+        this.examples = arrayOf(
+            "diabot quote search some funny words",
+            "diabot quote search r keyword"
+        )
+
+        this.maxNumberOfQuotes = System.getenv().getOrDefault("QUOTE_MAX_SEARCH_DISPLAY", "10").toInt()
+    }
+
+    override fun execute(event: CommandEvent) {
+        runBlocking {
+            launch {
+                if (!QuoteDAO.awaitCheckRestrictions(event.guildChannel, warnDisabledGuild = true)) return@launch
+
+                var args = event.message.contentRaw.split(" ").toList()
+                if (args.size < 3) {
+                    replyTooFewArgs(event)
+                    return@launch
+                }
+
+                // Remove the command portion of the message
+                args = args.slice(2..args.size - 1)
```

Comment:
[`until`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/until.html) might look cleaner?
```kotlin
args = args.slice(2 until args.size)
```

Up to you. IDEA has a warning about this which is why I'm proposing it.